.github/workflows/main_unit_tests.yml

```diff
@@ -0,0 +1,31 @@
+name: "Main unit tests"
```

Comment:
Is this workflow file needed?

Reply:
Are the tests contained within that directory also tested in CI? If so, it can be removed.